### PR TITLE
Benchmark and optimize japanese

### PR DESCRIPTION
**Pull request by ManyTheFish**
# Pull Request

- [x] Add benchmarks on Japanese texts
- [x] Fix benchmarks compilation
- [x] Conditionally detect the `Language` when there are several tokenizer choices for the same `Script`

### CODE REVIEW

File: src/analyzer.rs [hunk 1]
```diff
@@ -326,14 +326,36 @@ impl<'a, A> Analyzer<'a, A> {
...
+        match self
+            .config
             .pipeline_map
-            .get(&(script, language))
-            .or_else(|| self.config.pipeline_map.get(&(script, Language::Other)))
-            .or_else(|| self.config.pipeline_map.get(&(Script::Other, Language::Other)))
-            .unwrap_or_else(|| &*DEFAULT_PIPELINE)
+            .iter()
+            .filter(|((script, _), _)| *script == detected_script)
+            .collect::<Vec<_>>()[..]
```
Comment (Kerollmops) [inline src/analyzer.rs:335]: Could you do that in two lines, please? It would be clearer.

File: benches/bench.rs [hunk 2]
```diff
@@ -15,12 +15,16 @@ pub fn criterion_benchmark(c: &mut Criterion) {
...
+    ("364B_CHJP2", "本サイトで可能な検索は文字列検索（全文検索）だけですが、形態論情報を利用した検索サイト「中納言」も開設しています。詳しくは以下の URL をご覧下さい。なお、中納言の利用には申請が必要です。語学ないし文学の立場からの価値判断は一切おこなっていません"),
+    ("364B_HIJP2", "すもももももももものうち！「すもも」も「もも」も丸々としておいしそうという特徴で見ると「すもももももももものうち」は正しいことになります。今日は「すもももももももものうち」について考えます。今日は「すもももももももものうち」について考えます。"),
     ("363B_LTEN2", "The City of London Corporation is unique in the UK and has some unusual responsibilities for a local council, such as being the police authority. It is also unusual in having responsibilities and ownership beyond its boundaries. The Corporation is headed by the Lord Mayor of the City of London (an office separate from, and much older than, the Mayor of London)."),
     ("363B_LTFR1", "La position de Lutèce, sur l'île aujourd'hui nommée l'île de la Cité, permettant le franchissement du grand fleuve navigable qu'est la Seine par une voie reliant le Nord et le Sud des Gaules, en fait dès l'Antiquité une cité importante, capitale des Parisii, puis lieu de séjour d'un empereur romain. Le mot Lutèce resulte de la francisation de Lutetia."),
 ];
```
Comment (Kerollmops) [inline benches/bench.rs:30]: Are these the newly added benchmarks?

**Review by Kerollmops — CHANGES_REQUESTED** [review]

File: benches/bench.rs (hunk 2 above)
Comment (ManyTheFish) [reply benches/bench.rs:30]: Yes I added Japanese texts 😄

**Review by Kerollmops — APPROVED** [review]
Looks good to me mister Many!
bors merge

**Pull request merged**